packages/server/src/api/controllers/public/applications.ts

```diff
@@ -68,10 +69,32 @@ export async function destroy(ctx: any, next: any) {
   })
 }
 
+export async function unpublish(ctx: any, next: any) {
+  await doInAppContext(ctx.params.appId, async () => {
+    // get the app before deleting it
+    await controller.fetchAppPackage(ctx)
```

Comment:
Do we need this line? Like like we're ignoring the result

Reply:
I think a `204 - No Content` would be fine here, rather than the returning the app. 
Returning the published state of the app might not be correct, the status will show as published